renku/core/workflow/plan.py

```diff
@@ -132,3 +145,24 @@ def is_plan_removed(plan: AbstractPlan) -> bool:
             return True
 
     return False
+
+
+@inject.autoparams()
+def get_composite_plans_by_child(plan: AbstractPlan, plan_gateway: IPlanGateway) -> List[CompositePlan]:
+    """Return all composite plans that contain a child plan."""
+
+    derivatives = {p.id for p in get_derivative_chain(plan=plan)}
+
+    all_plans = plan_gateway.get_all_plans()
+
+    child_plan: Optional[AbstractPlan] = plan
+    while child_plan is not None:
+        derivatives.add(child_plan.id)
+        plan = child_plan
+        child_plan = next((p for p in all_plans if p.derived_from == plan.id), None)
```

Comment:
This `while` block is not needed since `get_derivative_chain` already returned all parents/children of the plan.